torch/_inductor/decomposition.py

```diff
@@ -484,15 +485,29 @@ def dequantize_per_tensor_tensor_decomp_impl(
     return (input.to(torch.float32) - zero_point) * scale
 
 
-@register_decomposition(torch.ops.quantized.embedding_bag_byte_unpack)
-def q_embedding_bag_byte_unpack_decomp(packed):
-    def bitcast_u8_to_f32(u8):
-        x, y, z, w = (u8[..., n].to(torch.int32) for n in (0, 1, 2, 3))
-        return (x + (y << 8) + (z << 16) + (w << 24)).view(torch.float32)[..., None]
+if sys.byteorder == "little":
```

Comment:
Can we move this check inside the function to reduce the duplicated code, something like:
```
if sys.byteorder == "little":
   return (x + (y << 8) + (z << 16) + (w << 24)).view(torch.float32)[..., None]
else:
   return ((x << 24) + (y << 16) + (z << 8) + w).view(torch.float32)[..., None]
```

Reply:
Thanks, I'll update it. I expected that it wouldn't work, but it actually did.